### DISCUSSION TITLE
Replacing map doesn't work properly, any thoughts on how to debug?

### DISCUSSION THREAD

**Discussion by hindenbugbite** [itzg/docker-minecraft-server#1202] [Q&A] [answered]
I've been running this docker for vanilla and paper servers without any issues before. Recently decided to try a downloaded adventure map: https://www.planetminecraft.com/project/under-olympus-a-puzzle-adventure-map-by-the-tater-team-java-1-17/

When I copy the map folder and replace the server.properties files as stated in the readme, the server starts and looked fine at first. But when I join the server, none of the in game buttons and levers work and I can't interact with objects. (when I pull a lever, it gives off some red smoke but doesn't toggle) I checked that command blocks are enabled in the properties. So far I've tried to spin up the docker in different ways, paper vs vanilla, using env to copy the world instead of manually copy, replacing vs editing the server.properties file. All of them end up with the same issue. 

I'm wondering if it could be some type of file permission issue when replacing the map? I'm hoping someone has some ideas of what debug message or test I can do to find out what's not working. I've also run the map as single player as a Minecraft saved game and that works fine.

**Comment by hindenbugbite** [comment, marked as answer]
I think I figured out what I was missing, it has to do with op permissions. Seems like all players need op to make things work on this map and I have a habit of not granting op to my kids. It also explains why local single player works because the default is op level 4, while server multiplayer default is level 0.

**Reply by itzg** [reply]
That makes sense. Thanks for following up.

**Reply by MaxamServices** [reply]
@hindenbugbite Im trying to do something similar, when setting up the container did you assigned an accessible folder to the docker volume? or how did you copied the new map's folder to the docker volume?

**Reply by hindenbugbite** [reply]
@MaxamServices  I do bind a folder outside the container to /data so I can the map files. Using a docker volume would also do the same thing, just need to find where the volume path is. (i.e. /var/lib/docker/volumes/)

I think either copy the new map overwriting the old map, or as a separate folder and edit server.properties to point to the new map folder.
